picard/album.py

```diff
@@ -409,7 +407,7 @@ def _finalize_loading(self, error):
             del self._new_tracks
             self.loaded = True
             self.status = None
-            self.match_files(self.unmatched_files.files)
+            self.match_files(unmatched_files + self.unmatched_files.files)
```

Comment:
I'd move `unmatched_files = [file for track in self.tracks for file in track.files]` here, and perhaps even drop the intermediate variable `unmatched_files` since we have no need to store this list.
It can be made a generator (dunno performance-wise):

```python
function unmatched_files():
    for track in self.tracks:
        for file in track.files:
            yield file
  
self.match_files(unmatched_files() + self.unmatched_files.files)
```

Reply:
No, we need to do it before the old tracks in self.tracks get dropped and overwritten with self._new_tracks, and we have to match files afterwards when _new_tracks have been placed in self.tracks. That's why before they got moved away to unmatched files. But that's not really necessary, holding them in a variable is enough. match_files will then move them to unmatched in case there is no match.

Reply:
Ok thanks for clarification, I overlooked that.